app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -6,7 +6,29 @@ def create_destination
   def determine_placement
     availability_zone = placement
     options[:dest_availability_zone] = [availability_zone.try(:id), availability_zone.try(:name)]
-    signal :prepare_provision
+    signal :prepare_volumes
+  end
+
+  def prepare_volumes
+    if options[:volumes]
+      phase_context[:requested_volumes] = create_requested_volumes(options[:volumes])
```

Comment:
The `create_requested_volumes` and `do_volume_creation_check` methods are only defined for Openstack but the method calls are here at the base cloud_manager/provision level.

We should provide base versions of these two methods that raise the `NotImplementedError` similar to what we do in [app/models/miq_provision/state_machine.rb#L8](https://github.com/ManageIQ/manageiq/blob/3ec71050868dd50cb2e9bb46b2e42a895bddbe51/app/models/miq_provision/state_machine.rb#L8) and elsewhere to avoid chasing down 'undefined method` errors if this code gets called unexpectedly.

Suggest creating a `volume_attachment.rb` file for this base level to contain the NotImplementedError methods.